### ISSUE TITLE
mvs-texturing doesn't build on macOS using default clang compiler toolchain - "Undefined symbols for architecture x86_64:   acc::BVHTree<unsigned int, math::Vector<float, 3> >::NAI"

### ISSUE THREAD

**Issue by mikeroberts3000** [nmoehrle/mvs-texturing#65]
Hello there,

When attempting to build mvs-texturing from scratch on macOS using the default clang compiler toolchain, I get the following error.

```
[100%] Linking CXX executable texrecon
cd /Users/mike/Code/GitHub/mvs-texturing/BUILD/apps/texrecon && /Applications/CMake.app/Contents/bin/cmake -E cmake_link_script CMakeFiles/texrecon.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -std=c++11 -DRESEARCH=1 -O2 -g -DNDEBUG -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/texrecon.dir/arguments.cpp.o CMakeFiles/texrecon.dir/texrecon.cpp.o  -o texrecon  -L/Users/mike/Code/GitHub/mvs-texturing/elibs/mve/libs/mve  -L/Users/mike/Code/GitHub/mvs-texturing/elibs/mve/libs/util  -L/Users/mike/Code/GitHub/mvs-texturing/BUILD/elibs/ext_gco/src/ext_gco-build -Wl,-rpath,/Users/mike/Code/GitHub/mvs-texturing/elibs/mve/libs/mve -Wl,-rpath,/Users/mike/Code/GitHub/mvs-texturing/elibs/mve/libs/util -Wl,-rpath,/Users/mike/Code/GitHub/mvs-texturing/BUILD/elibs/ext_gco/src/ext_gco-build  ../../libs/tex/libtex.a -lmve -lmve_util ../../libs/mrf/libmrf.a -lgco /usr/local/lib/libjpeg.dylib -framework libpng /usr/lib/libz.dylib /usr/local/lib/libtiff.dylib 
Undefined symbols for architecture x86_64:
  "acc::BVHTree<unsigned int, math::Vector<float, 3> >::NAI", referenced from:
      acc::BVHTree<unsigned int, math::Vector<float, 3> >::bsplit(unsigned int, std::__1::vector<acc::AABB<math::Vector<float, 3> >, std::__1::allocator<acc::AABB<math::Vector<float, 3> > > > const&) in libtex.a(calculate_data_costs.cpp.o)
      acc::BVHTree<unsigned int, math::Vector<float, 3> >::ssplit(unsigned int, std::__1::vector<acc::AABB<math::Vector<float, 3> >, std::__1::allocator<acc::AABB<math::Vector<float, 3> > > > const&) in libtex.a(calculate_data_costs.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [apps/texrecon/texrecon] Error 1
make[1]: *** [apps/texrecon/CMakeFiles/texrecon.dir/all] Error 2
make: *** [all] Error 2
```

Note that I executed make VERBOSE=1 in this example, so we can see the exact linker command that generate this error.

**Comment by nmoehrle** [comment]
This boils down to the same problem that causes #58 and the correct fix would be to finally replace [rayint](https://github.com/nmoehrle/rayint) with [libacc](https://github.com/nmoehrle/libacc). I'll try to to this next week or push a fix to rayint...

**Comment by mikeroberts3000** [comment]
Ah, no worries. Thanks for looking into this. As a hacky workaround, I just commented out the offending line of code and replaced it with a #define, similar to NAI is defined for KDTrees. This seems to be working correctly.

**Comment by vijayrajkumar** [comment]
Hi, I'm getting the same errors on MacOS. @mikeroberts3000 Could you possibly share in more detail how you were able to get the code to compile successfully?

**Comment by vijayrajkumar** [comment]
@nmoehrle do you have any suggestions on what I can do to get the code to compile properly (regarding this same error)? Do you plan on updating this repository soon? I would really love to use mvs-texturing for my research project as I need to accurately map registered images to a 3D model. Hope to hear from you soon. Thanks!

**Comment by mikeroberts3000** [comment]
@vijayrajkumar Just search the mvs-texturing folder for "NAI". It's defined using a #define in acc::KDTree, and it's defined with a static constexpr in acc::BVHTree. As far as I can tell, the two styles are functionally equivalent. But the static constexpr definition didn't compile on my system. So I just replaced the non-working static constexpr definition with the working #define style in acc::BVHTree.

Of course this is just a quick and dirty hack. I'm sure anything that @nmoehrle recommends will be better than what I'm recommending here.

**Comment by nmoehrle** [comment]
Sorry I didn't have time to properly fix this issue. Mikes solution is perfectly fine I just tried to get the constant into the namespace in case somebody else uses NAI as well...

The cleanest solution would be something similar to what I did here https://github.com/nmoehrle/libacc/blob/develop/bvh_tree.h#L121.
Hope that I will find the time to properly replace rayint with libacc sometime next month.
